docs/external-node.md

```diff
@@ -0,0 +1,568 @@
+   kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-cluster $ANTREA_CLUSTER_NAME --server=$ANTREA_API_SERVER --insecure-skip-tls-verify=true
+   kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-credentials antrea-agent --token=$TOKEN
+   kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-context antrea-agent@$ANTREA_CLUSTER_NAME --cluster=$ANTREA_CLUSTER_NAME --user=antrea-agent
+   kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig use-context antrea-agent@$ANTREA_CLUSTER_NAME
+   # Copy antrea-agent.antrea.kubeconfig to the VM
+   ```
+
+6. Create an `ExternalNode` resource for the VM.
+
+   After preparing the `ExternalNode` configuration yaml for the VM, we can
+   apply it in the cluster.
+
+   ```bash
+   $ cat << EOF | kubectl apply -f -
+   apiVersion: crd.antrea.io/v1alpha1
+   kind: ExternalNode
+   metadata:
+     name: vm1
+     namespace: vm-ns
+     labels:
+       role: db
+   spec:
+     interfaces:
+       - ips: [ "172.16.100.3" ]
+         name: ""
+   EOF
+   ```
+
+### Installation on Linux VM
+
+#### Prerequisites on Linux VM
+
+OVS needs to be installed on the VM. For more information about OVS installation
+please refer to the [getting-started guide](getting-started.md#open-vswitch).
+
+#### Installation steps on Linux VM
+
+1. Build `antrea-agent` binary in the root of the antrea code tree and copy the
+   `antrea-agent` binary from the `bin` directory to the Linux VM.
+
+   ```bash
+   make docker-bin
+   ```
+
+2. The `antrea-agent.conf` file specifies agent configuration parameters. Copy
+   the [agent configuration file](../build/yamls/externalnode/conf/antrea-agent.conf)
+   to the VM and edit the `antrea-agent.conf` file to set `clientConnection`,
+   `antreaClientConnection` and `externalNodeNamespace` with the correct values.
+   Copy `antrea-agent.antrea.kubeconfig` and `antrea-agent.kubeconfig` files to
+   the VM, that were generated in the step 4 and step 5 of
+   [Prerequisites on Kubernetes cluster](#prerequisites-on-kubernetes-cluster).
+
+   ```bash
+   AGENT_NAMESPACE="vm-ns"
+   AGENT_CONF_PATH="/etc/antrea"
+   mkdir -p $AGENT_CONF_PATH
+   # Copy antrea-agent kubeconfig files
+   cp ./antrea-agent.kubeconfig $AGENT_CONF_PATH
+   cp ./antrea-agent.antrea.kubeconfig $AGENT_CONF_PATH
+   # Update clientConnection and antreaClientConnection
+   sed -i "s|kubeconfig: |kubeconfig: $AGENT_CONF_PATH/|g" antrea-agent.conf
+   sed -i "s|#externalNodeNamespace: default|externalNodeNamespace: $AGENT_NAMESPACE|g" antrea-agent.conf
+   # Copy antrea-agent configuration file
+   cp ./antrea-agent.conf $AGENT_CONF_PATH
+   ```
+
+3. Configure environment variable `NODE_NAME` if the VM's hostname is different
+   from the name defined in the `ExternalNode` resource.
+
+   ```bash
+   export NODE_NAME="vm1"
```

Comment:
again, I am not sure how this can work for a systemd service. It doesn't feel like this environment variable will be available to the antrea-agent process.

Instead I would expect something like this in `/etc/systemd/system/antrea-agent.service`:
```
[Service]
Environment=NODE_NAME=vm1
ExecStart=...
```

But I am not an expert on this so I may be missing something...

Reply:
Since this part is copied from the original vm-installation.md,  @Anandkumar26 does this setting work for service? If yes, I would update it in this PR.

Reply:
Move the environment variable setting into the service configuration.

Reply:
We can update the environment variable this way.
```shell
[Service]
Environment="NODE_NAME=$NODE_NAME"
Environment="EXTERNAL_NODE=$EXTERNAL_NODE"
```
https://gitlab.eng.vmware.com/antreaplus/antreacloud/-/blob/master/build/images/agent/linux/install.sh#L527

Reply:
Variable "EXTERNAL_NODE" is not introduced in the feature branch